backend/src/http.rs

```diff
@@ -76,13 +78,32 @@ impl Service<Request<Body>> for RootService {
             req.uri().query().map(|q| format!("?{}", q)).unwrap_or_default(),
         );
 
+        /// Responds with the asset identified by the given name.
+        fn serve_asset(name: &str) -> <RootService as Service<Request<Body>>>::Future {
```

Comment:
This is already indented pretty deep; is there really an advantage to this being a local function?